File: leetcode/python/0119_pascals_triangle_II.py
class Solution(object):
    def getRow(self, rowIndex):
        """
        :type rowIndex: int
        :rtype: List[int]
        """
        if rowIndex == 0: return [1]
        if rowIndex == 1: return [1, 1]
        prev = [1, 1]
        for i in range(2, rowIndex + 1):
            res = []
            for j in range(0, i + 1):
                if j == 0 or j == i:
                    res.append(1)
                else:
                    res.append(prev[j - 1] + prev[j])
            prev = res

        return res

File: leetcode/java/0119_pascals_triangle_II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> prev = new ArrayList<Integer>(List.of(1, 1));
        List<Integer> curr = new ArrayList<Integer>();

        if (rowIndex == 0) return new ArrayList<Integer>(List.of(1));
        else if (rowIndex == 1) return prev;
        else {
            for (int i = 0; i <= rowIndex; i++) {
                curr = new ArrayList<Integer>();
                for (int j = 0; j <= i; j++) {
                    if (j == 0 || j == i) curr.add(1);
                    else curr.add(prev.get(j - 1) + prev.get(j));
                }
                prev = curr;
            }
        }

        return curr;
    }
}